Assets/Scripts/KatamariMovement.cs

```diff
@@ -342,22 +309,131 @@ void PowerBar()
         }
 
 
-        power_bar.GetComponent<Image>().fillAmount = power;
+        powerBar.GetComponent<Image>().fillAmount = power;
 
 
     }
 
     // Returns True if in golf hit mode
     public bool IsGolfHitMode()
     {
+
+        if (myStateMachine == stateMachine.golfMode)
+        {
+            isGolfHitMode = true;
+        } else
+        {
+            isGolfHitMode = false;
+        }
         return isGolfHitMode;
     }
```

Comment:
I think we have some state duplication now between the state machine and `isGolfHitMode` (the boolean member variable). We should be able to remove the boolean altogether at this point. I think this function is OK to keep for readability though.

```suggestion
    public bool IsGolfHitMode()
    {
        return myStateMachine == stateMachine.golfMode;
    }
```